dask/order.py

```diff
@@ -349,6 +351,14 @@ def dependencies_key(x):
         while item in result:
             item = init_stack_pop()
 
+    # Check task annotations
+    for k, v in dsk.items():
+        if istask(v):
```

Comment:
FYI, the order benchmarks from dask/dask-benchmarks don't show any slowdowns here. I thought that these two lines might be expensive, but if they are we don't catch it in those benchmarks.

Reply:
I am both pleased and surprised by this

Reply:
Oh, I wasn't trying to cast doubt on the benchmarks.  I'm genuinely pleased that this didn't have much of an effect.  I haven't really taken much of a look at the benchmarks to know what's going on in them to be able to say either way.  I'm mostly trusting that you and Erik have this under control.

Reply:
👍 

FWIW I tried a bit but couldn't make this block take more than ~3% or so of the run time. What's the worst-case scenario here? My thought was something like

```diff
diff --git a/dask/benchmarks/order.py b/dask/benchmarks/order.py
index 2268fe3..620d3f7 100644
--- a/dask/benchmarks/order.py
+++ b/dask/benchmarks/order.py
@@ -1,5 +1,6 @@
 from dask import array as da
 from dask.base import collections_to_dsk
+from dask.core import get_dependencies
 from dask.order import order
 from .common import DaskSuite
 
@@ -162,3 +163,19 @@ class OrderManySubgraphs(DaskSuite):
 
     def time_order_many_subgraphs(self, param):
         order(self.dsk)
+
+
+class TimeOrderMisc(DaskSuite):
+    def setup(self):
+        dsk = {'0': (0,)}
+        for i in range(1, 1_000_000):
+            dsk[str(i)] = (f, str(i - 1))
+
+        dependencies = {k: get_dependencies(dsk, k) for k in dsk}
+
+        self.dsk = dsk
+        self.dependencies = dependencies
+
+    def time_order_silly(self):
+        order(self.dsk, dependencies=self.dependencies)

```

1. A large task graph (since this is making another pass over `dsk.items()`
2. A very simple task graph (less time in the rest of the `order` code)

even with that, I get a slight slowdown (7.58s -> 8.2s)

Reply:
It wouldn't be the first time a benchmark proved our intuition wrong 😉